sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -831,15 +834,17 @@ object GpuCast extends Arm {
           child, elementType, StringType, ansiMode, legacyCastToString, stringToDateAnsiModeEnabled)
       }
 
-      withResource(strChildContainsNull) {strChildContainsNull =>
-        withResource(input.replaceListChild(strChildContainsNull)) {strArrayCol =>
-          withResource(concatenateStringArrayElements(strArrayCol, legacyCastToString)) {strCol =>
-            withResource(addBrackets(strCol)) {
-              _.mergeAndSetValidity(BinaryOp.BITWISE_AND, input)
-            }
-          }
-        }
+      val strArrayCol = withResource(strChildContainsNull) {
+        input.replaceListChild
+      }
+      val strCol = withResource(strArrayCol) {
+        concatenateStringArrayElements(_, legacyCastToString)
       }
+      val strColWithBrackets = withResource(strCol) {
+        addBrackets
+      }
+
+      withResource(strColWithBrackets)( _.mergeAndSetValidity(BinaryOp.BITWISE_AND, input))
```

Comment:
Style Nit: it's inconsistent to use a one-liner here but not on the simpler blocks above.

Reply:
I was reformatting based on @abellina's comment, missed this line. 

We should use scalafmt to streamline the review process.